tuplex/utils/src/Row.cc

```diff
@@ -177,13 +177,10 @@ namespace tuplex {
                         serializer.append(el.isNull() ? option<std::string>::none : option<std::string>(std::string((const char*)el.getPtr())), el.getType());
                     else if(rt.isListType())
                         serializer.append(el.isNull() ? option<List>::none : option<List>(*(List*)el.getPtr()), el.getType());
-                    else {
-                        if(rt.isTupleType()) {
-
-                            // needs specialized function, depending on tuple type...
-                            throw std::runtime_error(std::string(__FILE__) + " + " + std::to_string(__LINE__) + std::string(": tuple serialization not yet implemented"));
-                            //serializer.append(*((Tuple *)el.getPtr()));
-                        } else throw std::runtime_error("option underlying type " + rt.desc() + " not known");
+                    else if (rt.isTupleType()) {
+                        serializer.append(el.isNull() ? option<Tuple>::none : option<Tuple>(*(Tuple*)el.getPtr()), el.getType());
```

Comment:
add potentially assert st on the *(Tuple*)el.getPtr() expression so we don't deref a null pointer? Could lead to a pretty bad sigsev else